tools/editors-libraries.adoc

```diff
@@ -0,0 +1,177 @@
+<div id="dmn-editor-container"/>
+----
+
+Then load each Editor, passing the reference to its respective `div` container:
+
+[source,javascript,subs="+attributes"]
+----
+const bpmnEditor = BpmnEditor.open({
+ container: document.getElementById("bpmn-editor-container"), <1>
+ initialContent: Promise.resolve(""), <2>
+ readOnly: false, <3>
+ origin: "*", <4>
+ onError: () => {}, <5>
+ resources: new Map([]) <6>
+});
+
+const dmnEditor = DmnEditor.open({
+ container: document.getElementById("dmn-editor-container"), <1>
+ initialContent: Promise.resolve(""), <2>
+ readOnly: false, <3>
+ origin: "*", <4>
+ onError: () => {}, <5>
+ resources: new Map([]) <6>
+});
+----
+<1> `container`: HTML element to which the Editor will be appended to.
+<2> `initialContent`: Promise to a BPMN XML content for the BPMN Editor or the XML content of a DMN file for the DMN Editor. Can be empty. Examples:
+ * `Promise.resolve("")`
+ * `Promise.resolve("<BPMN_OR_DMN_CONTENT_DIRECTLY_HERE>")`
+ * `fetch("MyBpmnFile.bpmn").then(content => content.text())`
+<3> `readOnly` (optional, defaults to `true`): Use `false` to allow content edition and `true` for read-only mode, in which the Editor will not allow changes.
+<4> `origin` (optional, defaults to `*` when accessing the application with the `file://` protocol, `window.location.origin` otherwise): If the application needs to change this parameter, use it.
+<5> `onError` (optional, defaults to `() => {}`): If there's an error opening the Editor, this function will be called.
+<6> `resources` (optional, defaults to an empty Map): Map of resources that will be provided for the Editor. For instance, this parameter can be used to provide included models for the DMN Editor or Work Item Definitions for the BPMN Editor. Each entry in the map has the resource name as its key and an object containing the `content-type` (`text` or `binary`) and the resource `content` (Promise similar to the `initialContent` parameter) as its value. Example:
+ * `resources: new Map([["myWid.wid", { contentType: "text", content: Promise.resolve("<WID_CONTENTS_HERE>")}], [...], [...]])`
+ * `resources: new Map([["types.dmn", { contentType: "text", content: Promise.resolve("<DMN_CONTENTS_HERE>")}], [...], [...]])`
+
+The returned objects (`bpmnEditor` and `dmnEditor`) will contain the methods needed to manipulate the Editor:
+
+* `getContent(): Promise<string>`: Returns a Promise containing the Editor's content.
+* `setContent(normalizedPosixPathRelativeToTheWorkspaceRoot: string, content: string): Promise<void>`: Sets the content of the editor. The returning Promise will be rejected if setting the content fails.
+* `getPreview(): Promise<string>`: Returns a Promise containing the SVG string of the current diagram.
+* `subscribeToContentChanges(callback: (isDirty: boolean) => void): (isDirty: boolean) => void`: Setup a callback to be called on every content change in the Editor. Returns the same callback to be used for unsubscription.
```

Comment:
Setup a callback to be called on every content change in the Editor -> This will set up a callback to......?